docs/actions/check.rst

```diff
@@ -3,10 +3,15 @@ Check
 
 .. note::
     The logic for whether checks will be added by default is as follows:
-    1. If no action is provided in either pass, fail or error , add `checks` as default (to be backward compatible)
-    2. If only actions other than `checks` is provided, don't add check as default (to support cases where checks are not wanted)
-    3. If checks is a part of the actions provided, all pass , fail and error cases will have `checks` (to prevent case where a check is hanged and never finish processing)
+    1. If no action is provided in either pass, fail or error, add ``checks`` as default (to be backward compatible)
 
+    2. If only actions other than ``checks`` is provided, don't add check as default (to support cases where checks are not wanted)
+
+    3. If checks is a part of the actions provided, all pass, fail and error cases will have `checks` (to prevent case where a check is hanged and never finish processing)
+::
+
+.. warning::
+    Failing to add ``checks`` when no other action is provided (see point 1. above) will result in the checks not being executed. You must explicitly add them as a default.
```

Comment:
This should not be the case, `checks` should be added as default if `action` is not provided.

Reply:
I can see this occurring if you have empty `action` section